Roll up (#65)
cbor-wg/edn-literal: draft-ietf-cbor-edn-literals.md

> * `cddl` (which is used for the Concise Data Definition Language,
>   CDDL, see {{terminology}} below).

> Note that EDN does not have a monopoly on text-based representation of
> CBOR data items.

Reviewer: ```suggestion
Note that EDN is not designed to be the only text-based representation of
CBOR data items.
```

For instance, {{YAML}} {{-yaml-media-type}} is able to represent most CBOR
data items, possibly requiring use of YAML's extension points.

> YAML does not provide certain features that can be useful with tools

Reviewer: it might be good to name these features or omit this

Author: Done (not all features, just two examples).

instance, the following are equivalent:

~~~~ cbor-diag
   h'48656c6c6f20776f726c64'
   h'48 65 6c 6c 6f 20 77 6f 72 6c 64'
   h'4 86 56c 6c6f
     20776 f726c64'
~~~~

Note that the internal syntax of prefixed single-quote literals such
as h'' and b64'' can allow comments as blank space (see {{comments}}).
Since slash characters are allowed in b64'', only inline comments are
available in b64 string literals.

~~~~ cbor-diag
   h'68656c6c6f20776f726c64'
   h'68 65 6c /doubled l!/ 6c 6f # hello
     20 /space/
     77 6f 72 6c 64' /world/
~~~~

### Embedded CBOR and CBOR Sequences in Byte Strings {#embedded-cbor-and-cbor-sequences-in-byte-strings}

Where a byte string is to carry an embedded CBOR-encoded item, or more
generally a sequence of zero or more such items, the diagnostic
notation for these zero or more CBOR data items, separated by commas,
can be enclosed in `<<` and `>>` to notate the byte string
resulting from encoding the data items and concatenating the result.
For
instance, each pair of columns in the following are equivalent:

~~~~ cbor-diag
   <<1>>              h'01'
   <<1, 2>>           h'0102'

>    <<"edn", null>>    h'63 65646E F6'

Reviewer: ```suggestion
   <<"hello", null>>    h'63 68656c6c6f f6'
```

Reviewer: using the "edn" might be more confusing here... also, is there a preferred casing for hex strings?

feel free to dismiss this suggestion.

Author: > is there a preferred casing for hex strings?

I prefer lower case.  Unfortunately, Section 8 of RFC 4648 uses upper case (while sprinkling a "case-insensitive" on the unhappy result, I might add) and that is mostly hardwired in various CBOR specs now.  RFC 6991 is explicitly lowercase.

* We should have an ART-wide agreement to move to lowercase, where reasonable
* We need to think how much we can pick this up here.
